crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension.rs

```diff
@@ -118,13 +117,32 @@ pub(crate) fn unnecessary_dict_comprehension(
     let [target_key, target_value] = elts.as_slice() else {
         return;
     };
-    if ComparableExpr::from(key) != ComparableExpr::from(target_key) {
+
+    let Expr::Name(ast::ExprName {
+        id: target_key_name,
+        ..
+    }) = &target_key
+    else {
         return;
-    }
-    if ComparableExpr::from(value) != ComparableExpr::from(target_value) {
+    };
+    let Expr::Name(ast::ExprName {
+        id: target_value_name,
+        ..
+    }) = &target_value
+    else {
+        return;
+    };
+
+    let Expr::Name(ast::ExprName { id: key_name, .. }) = &key else {
+        return;
+    };
+    let Expr::Name(ast::ExprName { id: value_name, .. }) = &value else {
         return;
+    };
```

Comment:
Should I rewrite this to:

```rust
    let (
        Expr::Name(ast::ExprName {
            id: target_key_name,
            ..
        }),
        Expr::Name(ast::ExprName {
            id: target_value_name,
            ..
        }),
        Expr::Name(ast::ExprName { id: key_name, .. }),
        Expr::Name(ast::ExprName { id: value_name, .. }),
    ) = (&target_key, &target_value, &key, &value)
    else {
        return;
    };
```

?

Reply:
I think it's fine as is. You could possibly merge the `target_key_name` and `target_value_name` above like:

```rust
    let [Expr::Name(ast::ExprName { id: target_key, .. }), Expr::Name(ast::ExprName { id: target_value, .. })] = elts.as_slice() else {
        return;
    };
```